workflows/dc6-workflow.yaml

```diff
@@ -0,0 +1,1134 @@
+                key: azurestoragekey
+        command: [ dodola ]
+        args:
+          - "rechunk"
+          - "{{ inputs.parameters.in-zarr }}"
+          - "--out"
+          - "{{ inputs.parameters.out-zarr }}"
+          - "--chunk"
+          - "time={{ inputs.parameters.time-chunk }}"
+          - "--chunk"
+          - "lat={{ inputs.parameters.lat-chunk }}"
+          - "--chunk"
+          - "lon={{ inputs.parameters.lon-chunk }}"
+          - "--chunk"
+          - "dayofyear={{ inputs.parameters.dayofyear-chunk }}"
+        resources:
+          requests:
+            memory: 32Gi
+            cpu: "1000m"
+          limits:
+            memory: 32Gi
+            cpu: "2000m"
+      activeDeadlineSeconds: 10800
+
+
+    - name: apply-adjustmentfactors
+      inputs:
+        parameters:
+          - name: in-zarr
+          - name: yclimofine-zarr
+          - name: train-variable
+          - name: out-variable
+          - name: out-zarr
+      outputs:
+        parameters:
+          - name: out-zarr
+            value: "{{ inputs.parameters.out-zarr }}"
+      script:
+        image: downscalecmip6.azurecr.io/dodola:dev
+        imagePullPolicy: Always
+        env:
+          - name: IN_ZARR
+            value: "{{ inputs.parameters.in-zarr }}"
+          - name: CLIMO_ZARR
+            value: "{{ inputs.parameters.yclimofine-zarr }}"
+          - name: TRAIN_VARIABLE
+            value: "{{ inputs.parameters.train-variable }}"
+          - name: OUT_VARIABLE
+            value: "{{ inputs.parameters.out-variable }}"
+          - name: OUT_ZARR
+            value: "{{ inputs.parameters.out-zarr }}"
+          - name: AZURE_STORAGE_ACCOUNT_NAME
+            valueFrom:
+              secretKeyRef:
+                name: workerstoragecreds-secret
+                key: azurestorageaccount
+          - name: AZURE_STORAGE_ACCOUNT_KEY
+            valueFrom:
+              secretKeyRef:
+                name: workerstoragecreds-secret
+                key: azurestoragekey
+        command: [ python ]
+        source: |
+          import os
+          import xarray as xr
+          from skdownscale.spatial_models import SpatialDisaggregator
+
+          in_store_path = os.environ.get("IN_ZARR")
+          climo_path = os.environ.get("CLIMO_ZARR")
+          train_variable = os.environ.get("TRAIN_VARIABLE")
+          out_variable = os.environ.get("OUT_VARIABLE")
+          out_path = os.environ.get("OUT_ZARR")
+
+          af = xr.open_zarr(in_store_path)
+          print(f"read {in_store_path}")
+          obs_climo_fine = xr.open_zarr(climo_path)
+          print(f"read {climo_path}")
+
+          sd = SpatialDisaggregator(var=train_variable)
+          out = model.predict(af, obs_climo_fine[train_variable], var_name=train_variable)
+          print("spatial disaggregation adjustment factors applied")
+
+          out.to_dataset(name=out_variable).to_zarr(out_path, mode="w")
+          print(f"written to {out_path}")
```

Comment:
looks like we aren't actually writing to a zarr here? any reason for that?

Reply:
Eh? `out.to_dataset(name=out_variable).to_zarr(out_path, mode="w")` has a `.to_zarr()` on the end. I just chained the methods together on a single line of code. This is dumping to a zarr store unless I missed something.

Reply:
Ugh I missed that 😩